modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/EventInput.js

```diff
@@ -57,6 +62,13 @@ function EventInput({
 		(operator) => operator.name === criterion.operatorName
 	)?.name;
 
+	const showDateSelector = [
+		DATE_OPERATORS.BETWEEN,
+		RELATIONAL_OPERATORS.EQ,
+		RELATIONAL_OPERATORS.GT,
+		RELATIONAL_OPERATORS.LT,
+	].includes(criterion.day.operatorName);
```

Comment:
Hello @p2kmgcl! The day field will be undefined in the 'ever' date modifier so, accessing day.x might fail.